### PR TITLE
E2e start

### PR DESCRIPTION
**Pull request by bengerman13**
## Changes proposed in this pull request:

- Add e2e test framework
- Add a test to the e2e framework
- a tiny bit of pipeline cleanup

## Security considerations

This drives users around in dev, which could break stuff if done poorly.
We're also now storing MFA seeds (for unprivileged users in dev, which is only available via VPN) plaintext, which kinda defeats the purpose.

### CODE REVIEW

File: ci/pipeline.yml [hunk 1]
```diff
@@ -94,6 +94,48 @@ jobs:
...
+        image_resource:
+          type: docker-image
+          source:
+            repository: python
+            tag: "3.8"
+        run:
+          path: src/ci/e2e.sh
+        params:
+          KIBANA_URL: ((kibana-url))
+          DEV_TEST_USER_1_USERNAME: ((dev-test-user-1-username))
```
Comment (pburkholder) [inline ci/pipeline.yml:128]: @bengerman13 Did these commits trigger `caulking`? I should _not_ trigger, as I think there's a rule that .yml files with PASSWORD followed by '((...))' are NOT secrets.  

If you are getting alerts, let me know and I can fix that.

Comment (bengerman13) [reply ci/pipeline.yml:128]: it did not, which pleasantly surprised me :)

File: e2e/conftest.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+@pytest.fixture()
+def kibana_url():
+    return getenv("KIBANA_URL")
+
+
+@pytest.fixture
+def user_1():
+    user_1_username = getenv(f'DEV_TEST_USER_1_USERNAME')
+    user_1_password = getenv(f'DEV_TEST_USER_1_PASSWORD')
```
Comment (bengerman13) [inline e2e/conftest.py:15]: @pburkholder since you asked, though, these lines _did_ trigger caulking, which seemed weird.

**Review by onelittlebecca — APPROVED** [review]
LGTM

**Pull request merged**